### PR TITLE
Finished Github Searcher homework - react2-week1

### PR DESCRIPTION
**Pull request by RodrigoGuinsbergPinto**

### CODE REVIEW

File: react/react2/week1/app/src/GithubContext.jsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+const GithubProvider = ({ children }) => {
+  const [query, setQuery] = useState("");
+  const [users, setUsers] = useState([]);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState("");
+
+  const getInput = (e) => {
+    const value = e.target.value;
+    setQuery(value);
```
Comment (Divyajg) [inline react/react2/week1/app/src/GithubContext.jsx:13]: For more specificity, You can `setQuery()` after validating the value.

File: react/react2/week1/app/src/GithubContext.jsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+          const response = await fetch(
+            `https://api.github.com/search/users?q=${query}`
+          );
+          const data = await response.json();
+          setUsers(data.items);
+          setLoading(false);
+        } catch (error) {
+          setLoading(false);
+          setError(`💥 Error fetching: ${error.message}`);
+        }
```
Comment (Divyajg) [inline react/react2/week1/app/src/GithubContext.jsx:33]: Though `setLoading()` works inside try() or catch(), it is suggested to use another method finally(), where you can do setLoading(). [for your reference](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch)

File: react/react2/week1/app/src/components/UserSearch.jsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  const context = useContext(GithubContext);
+  const { query, users, loading, getInput } = context;
+
+  return (
+    <div>
+      <input
+        type="text"
+        value={query}
+        placeholder="insert user"
+        onChange={getInput}
```
Comment (Divyajg) [inline react/react2/week1/app/src/components/UserSearch.jsx:14]: You are taking event (e) as an argument to getInput(), you should be passing it in here. though it works sometimes, we should follow some rules while coding.

**Review by Divyajg — APPROVED** [review]
Very good work. ⭐

**Comment by RodrigoGuinsbergPinto** [comment]
HI Divyajg, thank you for your comments.
I will work on the implementation of my code and update this PR.

**Pull request merged**